### ISSUE TITLE
Associate Sources with Collaboration Papers

### ISSUE THREAD

**Issue by profjsb** [skyportal/skyportal#4238]
Allow users to publically "express an interest" in a source, see what papers in progress (or published) are associated with sources (and vice versa). This is a request from the LS4 collaboration -- this is a feature in another marshal and allows people across a large collaboration to know what sources are actively being worked on and who might be interested in helping.

**Comment by mcoughlin** [comment]
@profjsb Any screen shots or similar to give ideas on what an implementation might look like?

**Comment by profjsb** [comment]
yes, but i don't have permission to post those yet. Anyway, I want to do things a bit different. Will be happy to drive this one.

**Comment by mcoughlin** [comment]
Ok!

**Comment by Theodlz** [comment]
@profjsb @mcoughlin reviving this. Do we want any permissions on top of that, or would all the entries be visible to all users? I can see how we might want published papers to be published, but those being written only shared to specific folks or groups.

What do you think? This looks easy enough to implement, I think we just want:
- title
- description
- link
- share_group_ids

**Comment by Theodlz** [comment]
Coupling this with a quick retrieval button just like we do for the GCN circulars (with some chatGPT + RAG, and/or some proper code on our end to fetch specific services programmatically) should be pretty easy to implement.

**Comment by profjsb** [comment]
We need to think about the data model for this. We'll have the notion of a Publication/external communication. Each publication can be associated with 0 or more sources. Each publication will have 1 or more "owners" and 0 or more participants. Activity on a source associated with a publication should be part of the notification framework. 

We might consider enum attributes like "pub_type" and publication status. Will want a field for doi, url, abstract.

**Comment by mcoughlin** [comment]
@antoine-le-calloch note this is similar to the "Interested" request.
